plugins/woocommerce/client/blocks/assets/js/data/utils/is-editor.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * External dependencies
+ */
+import { getPath } from '@wordpress/url';
+
+/**
+ * Returns true if the current page is in the editor.
+ */
+export const isEditor = (): boolean => {
```

Comment:
I'm curious if there isn't already another way used to detect this. I saw this approach https://github.com/woocommerce/woocommerce/blob/7a636875210947e899b23675402097229245b285/plugins/woocommerce/client/blocks/assets/js/data/payment/utils/check-payment-methods.ts#L46 but I think we should check for `core/edit-post` as well. WDYT?

Reply:
That was my first option but ended up being too early to call it in a data store, hence why I switched to this static link approach which I'm not 100% confident about. If a frontend checkout page has `post.php` in it, which is something I never saw, then it would break.

Reply:
Not 100% about this one.

Could we pass something from server instead?

Reply:
We can pass something from the server however the server logic is still basically the same, this is how we check for siteEditor in server
```
'site-editor.php' === $pagenow
```

The logic I added matches was Gutenberg uses https://github.com/WordPress/gutenberg/blob/bb9272766d1ef923f9cbb12013ffdd2788a08cb7/packages/edit-post/src/deprecated.js#L24-L25

Reply:
I was thinking rather than targeting the editor we could target WP Admin which is easier to detect. That shouldn't impact the frontend store.